web/packages/test-helpers/src/generateBeefyCheckpoint.ts

```diff
@@ -50,23 +47,42 @@ async function generateBeefyCheckpoint() {
     const validatorSetId = await api.query.beefy.validatorSetId<ValidatorSetId>()
     const authorities = await api.query.beefy.authorities<BeefyId[]>()
 
+    console.log("validatorSetId:", validatorSetId)
+    console.log("authority length is:" + authorities.length)
+
     let addrs = []
+
     for (let i = 0; i < authorities.length; i++) {
-        let publicKey = publicKeyConvert(authorities[i], false).slice(1)
-        let publicKeyHashed = createKeccakHash("keccak256").update(Buffer.from(publicKey)).digest()
-        addrs.push(publicKeyHashed.slice(12))
+        console.log("index is:" + i + ",authority is:" + authorities[i])
+        try {
+            let publicKey = publicKeyConvert(authorities[i], false).slice(1)
+            let publicKeyHashed = createKeccakHash("keccak256")
+                .update(Buffer.from(publicKey))
+                .digest()
+            addrs.push(publicKeyHashed.slice(12))
```

Comment:
Wont this change the length of the list of validator addresses, which in turn will affect the merkle tree and eventual root? If the same logic is applied in the relayer then I think its fine.

Reply:
As the title mentioned, we don't manually construct the merkle root from the addresses as before, instead get `currentAuthorities` from runtime storage. The printing here is only for debug purpose.